datalad/support/tests/test_annexrepo.py

```diff
@@ -2194,6 +2199,60 @@ def test_annexjson_protocol(path):
     assert_not_in('errors from JSON records', str(e.exception))
 
 
+@with_tempfile
+def test_annexjson_protocol_long(path):
+    # Inform about the seed here and use dedicated RNG so we are not
+    # effected by any other use of random before this test
+    seed = _seed if _seed is not None else random.randint(0, 1000)
+    # Inform for ease of debugging.
+    # TODO: if other test like this appears, craft @with_rng helper
+    print("RNG seed: %s" % seed)
+    rng = random.Random(seed)
+    records = [
+        {('field%d' % f): ("a" * rng.randint(0, 5000))
+             for f in range(rng.randint(1, 100))
+         }
+        for i in range(10)
```

Comment:
I think it'd be better to avoid using random lengths here.  Without the annexrepo changes from this PR, this test failed for me only 6 out of the 10 times I ran it.

It looks like `Lib.asyncio.unix_events._UnixReadPipeTransport` reads with a max size of `256 * 1024`.  So perhaps something like this instead:

```diff
diff --git a/datalad/support/tests/test_annexrepo.py b/datalad/support/tests/test_annexrepo.py
index d04715122c..a5b7fedc75 100644
--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -2201,18 +2201,11 @@ def test_annexjson_protocol(path):

 @with_tempfile
 def test_annexjson_protocol_long(path):
-    # Inform about the seed here and use dedicated RNG so we are not
-    # effected by any other use of random before this test
-    seed = _seed if _seed is not None else random.randint(0, 1000)
-    # Inform for ease of debugging.
-    # TODO: if other test like this appears, craft @with_rng helper
-    print("RNG seed: %s" % seed)
-    rng = random.Random(seed)
     records = [
-        {('field%d' % f): ("a" * rng.randint(0, 5000))
-             for f in range(rng.randint(1, 100))
-         }
-        for i in range(10)
+        {"k": "v" * 20},
+        # Value based off of
+        # Lib.asyncio.unix_events._UnixReadPipeTransport.max_size.
+        {"k": "v" * 256 * 1024}
     ]
     with open(path, 'w') as f:
         for record in records:
```

Reply:
ok, will take yours but will compliment with bigger and shorter ones... just to make sure

Reply:
> just to make sure

Of course that's fine, though I'm not quite getting how that'd make you more sure than reverting your changes and seeing that it does reliably fail without the changes from this PR (as opposed to the, say, 6/10 I saw with the current version of the test).  IIRC from my local testing, the tests start to pass without the `AnnexJsonProtocol` changes from this PR at `(256 * 1024) - 9` (with 9 coming from the json quotes, etc).